src/plugins/drawing-tool/components/drawing-layer.tsx

```diff
@@ -189,6 +193,18 @@ export class DrawingLayerView extends React.Component<DrawingLayerViewProps, Dra
     let moved = false;
     const {selectedObjects, hoverObject } = this.state;
     let { objectsBeingDragged } = this.state;
+
+    // Prevents dragging multiple objects when you have one (or more) selected
+    // and then drag one unless shift/cmd pressed.
+    const selectedObjectsLength = this.state.selectedObjects.length;
+    if (selectedObjectsLength > 0 && !(e.shiftKey || e.metaKey)){
+      const clearSelectedObjects = this.state.selectedObjects;
+      for (let i = 0; i < selectedObjectsLength; i++){
+        clearSelectedObjects.pop();
+      }
+      this.setState((prevState)=> ({selectedObjects: clearSelectedObjects}));
```

Comment:
Is this doing more than:
```suggestion
      this.setState(prevState => ({ selectedObjects: [] }));
```
?

Reply:
Rather than add another place in the code that modifies the selection. It would be better to add this modifier logic to the place that is already updating the selection on drag. If you search for the `setSelectedObjects` you'll find 3 places that are modifying the selection. The place that is handling selection on drag is just below this part of the code. `handleSelectedObjectMouseDown` adds a `handleMouseMove` listener and that uses the `needToAddHoverToSelection` variable to decide if it should modify the selection.

`needToAddHoverToSelection` is currently true if hoverObject is not selected:
`if (hoverObject && !selectedObjects.some(object => object.id === hoverObject.id)) {`

I think the value of `needToAddHoverToSelection` is fine as is, that doesn't need to vary depending on the modifier key.  However inside that same if statement is where the `objectsToInteract` is set. Those are the objects that are being moved. The modifier key should change that behavior. If the hoverObject is selected then the current logic is fine. If the hoverObject isn't selected, and the modifier key is down, then the current logic is also correct `objectsToInteract = [hoverObject, ...selectedObjects]`. If the hoverObject isn't selected and modifier key is not down, then then it should be `objectsToInteractive = [hoverObject]`.

When `needToAddHoverToSelection` is used in handleMouseMove, if `needToAddHoverToSelection` is true then `this.setSelectedObjects(objectsToInteract)`. I think this should be fine as given the logic above.